### PR TITLE
Process data in models

### PR DESCRIPTION
**Pull request by laiifuu**
## Hi :wave: 
For this PR, I: 
- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Created custom methods for each of the model classes.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  after_create :update_comments_counter
+
+  belongs_to :user
```
Comment (elisha2kyakpo1) [inline app/models/comment.rb]: - [ ] The ERD diagram shows that the user should be named `author` which should give our relationship foreign key ID as `author_id` and not the default `user_id`.

Hint: `belongs_to :author, class_name: 'User'`

author

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  after_create :update_likes_counter
+
+  belongs_to :user
```
Comment (elisha2kyakpo1) [inline app/models/like.rb]: - [ ] The ERD diagram shows that the user should be named `author` which should give our relationship foreign key ID as `author_id` and not the default `user_id`.

author

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  after_create :update_posts_counter
+
+  belongs_to :user
```
Comment (elisha2kyakpo1) [inline app/models/post.rb]: - [ ] The ERD diagram shows that the user should be named `author` which should give our relationship foreign key ID as `author_id` and not the default `user_id`.

author

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (elisha2kyakpo1) [inline app/models/user.rb]: - [ ] In order to complete the relationship, we can do something like this below:
```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @laiifuu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 4 above)
Comment (laiifuu) [reply app/models/user.rb]: Hi @elisha2kyakpo1 :wave:, could you please explain what `foreign key: 'author_id` does?

Comment (Deepakdanger) [reply app/models/user.rb]: Hi @laiifuu , here `foreign key: 'author_id` is used to resemble user_id.

**Review by Deepakdanger — APPROVED** [review]
Hi @laiifuu , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**